src/client.js

```diff
@@ -25,29 +25,45 @@ class Client {
   /**
    * Primary entry point for building a new Client.
    */
-  static buildClient(config) {
+  static buildClient(config, fetchFunction) {
     const newConfig = new Config(config);
 
-    return new Client(newConfig);
+    return new Client(newConfig, GraphQLJSClient, fetchFunction);
   }
 
   /**
    * @constructs Client
    * @param {Config} config An instance of {@link Config} used to configure the Client.
    */
-  constructor(config, GraphQLClientClass = GraphQLJSClient) {
-    const apiUrl = `https://${config.domain}/api/graphql`;
+  constructor(config, GraphQLClientClass = GraphQLJSClient, fetchFunction) {
+    const url = `https://${config.domain}/api/graphql`;
 
-    this.graphQLClient = new GraphQLClientClass(types, {
-      url: apiUrl,
-      fetcherOptions: {
-        headers: {
-          'X-SDK-Variant': 'javascript',
-          'X-SDK-Version': version,
-          'X-Shopify-Storefront-Access-Token': config.storefrontAccessToken
+    const headers = {
+      'X-SDK-Variant': 'javascript',
+      'X-SDK-Version': version,
+      'X-Shopify-Storefront-Access-Token': config.storefrontAccessToken
+    };
+
+    if (fetchFunction) {
+      headers['Content-Type'] = 'application/json';
+      headers.Accept = 'application/json';
+
+      this.graphQLClient = new GraphQLClientClass(types, {
+        fetcher: function fetcher(graphQLParams) {
```

Comment:
What about modifying `httpFetcher` to accept a `fetchFunction` as well?

graphql-js-client has the same issue right now I assume? It would allow both to be fixed.

Reply:
I thought about that, but I think I like graphql js client to be more open. It accepts an `httpFetcher` function that just receives GraphQL params. I think in the wider node world, we shouldn't assume `fetch` as an API, but in this one library it's probably a safer, but still less-ideal hedge.